### ISSUE TITLE
Include sites-enabled in nginx.conf

### ISSUE THREAD

**Issue by mikey0000** [rockstor/rockstor-core#2129]
I've got a custom nginx.conf which has added this line.

include /opt/rockstor/etc/nginx/sites-enabled/*.conf;

and server names.

every time rockstor updates it nukes my changes so figured adding it should be the better option. Can PR, not sure how to address adding multiple domain names, is this in the docs?

**Comment by phillxnet** [comment]
@mikey0000 I'm not sure this has a place in Rockstor's core code. Our Nginx server is part of the 'appliance' side of things and I don't want it to become a generic web server config as that would constitute mission creep. I see that if it's running one might want to use it but I think the Appliance / Rockstor way to have a configurable web server would be to use something like this pending rock-on:

https://github.com/rockstor/rockon-registry/pull/155

That way you would have complete isolation of your desired nginx server and it would not be affected by any Rockstor updates. Plus it would be the latest official nginx.

Take a look at the pr and see what you think. I requested some changes to it just recently but I suspect the negligence on my part with that pull request has see @JasonTarka interest wain. If we get no response soon then maybe you could take that prs last state and add my changes and present it as a new pr. @magicalyak may also step up again and present their turn-key linuxserver.io based rock-on. That definitely seems more the way to go to me as I'm not keen on maintaining any secondary use of our 'appliance' nginx config. Especially as in the futue we may well be applying some rather specific configuration to it and if folks are expecting it to be accessible to general use that is not going to work out.

So in short this looks like a job for a Rock-on and unfortunately I dropped the ball on the very one that would server your needs. But if you are happy to chip in and credit @JasonTarka for initiating the initiative, linking to that pr, we might get such a facility up and running pretty quick.

I've updated that nginx Rock-on pull request again in the hope we can entice @JasonTarka to apply my suggestions and linked back to this issue. But it's been quite some time since they were last actively involved in it. Feel free to start another pull request in that repo if this interest you.

**Comment by FroggyFlox** [comment]
> So in short this looks like a job for a Rock-on

@mikey0000 , following up on this comment, maybe the recently added Nginx Proxy Manager might be of interest to you. See the forum post linked below for some resources:
https://forum.rockstor.com/t/rock-ons-changelog/6469/11

**Comment by mikey0000** [comment]
Thanks for the quick response, I'll take a look today.

**Comment by phillxnet** [comment]
@mikey0000 Just a notification that the previously linked generic official nginx docker Rock-on has now been merge and published, thanks to @JasonTarka resurfacing after the pull request suffered from my neglect. I believe that @magicalyak is also likely to spin us up a linuxserver.io variant that should make for a nice complement to the bare and no-config-by-default existing nginx Rock-on.

**Comment by phillxnet** [comment]
@mikey0000 Just revisiting this issue. I'm proposing that we now close this issue as we now have the previously referenced nginx container in play. And I'd like to keep 'our' nginx config confined to just core Rockstor functions to try and keep things simple.

Are you happy to close this now we have the rock-on way to achieve an independent nginx which is obviously then completely configurable.

**Comment by mikey0000** [comment]
yes I'll close it now.
